src/Tests/Microsoft.TemplateEngine.Cli.UnitTests/ParserTests/InstantiateTests.cs

```diff
@@ -258,7 +258,12 @@ internal void CanParseNameOption(string command, string? expectedValue)
             InstantiateCommandArgs args = InstantiateCommandArgs.FromNewCommandArgs(new NewCommandArgs(myCommand, parseResult));
             TemplateCommand templateCommand = new(myCommand, settings, packageManager, templateGroup, templateGroup.Templates.Single());
             CliConfiguration parser = ParserFactory.CreateParser(templateCommand);
-            ParseResult templateParseResult = parser.Parse(args.TokensToInvoke ?? Array.Empty<string>());
+            var tokensToInvoke = args.TokensToInvoke ?? Array.Empty<string>();
+            if (tokensToInvoke.Last() == "foo")
+            {
+                tokensToInvoke = tokensToInvoke.SkipLast(1).ToArray();
+            }
+            ParseResult templateParseResult = parser.Parse(tokensToInvoke);
```

Comment:
This is fun as `foo` would previously be 'unparsed' and the test would run properly. Now, it was getting picked up as the name argument and thus having 2 assigned names, which we've made into an error.